Source/Localhost/LocalhostServer.swift

```diff
@@ -222,3 +254,59 @@ extension CRRequest {
     }
 }
 
+struct LocalhostServerMethodPath: Hashable, Equatable {
+    let method: String
+    let path: String
+    
+    func hash(into hasher: inout Hasher) {
+        hasher.combine(method)
+        hasher.combine(path)
+    }
+    
+    public static func ==(lhs: LocalhostServerMethodPath, rhs: LocalhostServerMethodPath) -> Bool {
+        guard lhs.method == rhs.method else { return false }
+        guard lhs.path == rhs.path else { return false }
+        return true
+    }
+}
+
+struct LocalhostServerRoute {
+    let pathMethod: LocalhostServerMethodPath
+    let routeBlock: ((URLRequest) -> LocalhostServerResponse?)
+}
+
+extension LocalhostServer {
+    func setOverlayingRoute(method: String,
+                            path: String,
+                            routeBlock: @escaping ((URLRequest) -> LocalhostServerResponse?)) {
+        
+        let localhostServerMethodPath = LocalhostServerMethodPath(method: method, path: path)
+        let newServerRoute = LocalhostServerRoute(pathMethod: localhostServerMethodPath, routeBlock: routeBlock)
+        let methodPathKeys = Array(self.overlayingRoutes.keys)
+        if methodPathKeys.contains(localhostServerMethodPath), var existingRoutes = self.overlayingRoutes[localhostServerMethodPath] {
+            existingRoutes.append(newServerRoute)
+            self.overlayingRoutes[localhostServerMethodPath] = existingRoutes
```

Comment:
Could be 
```
if methodPathKeys.contains(localhostServerMethodPath),
     _ = self.overlayingRoutes[localhostServerMethodPath] {
    self.overlayingRoutes[localhostServerMethodPath].append(newServerRoute)
}
```
But that's just an opinion.

Reply:
Is this a trick question? Lol... 

When dealing with a dictionary of structs, accessing the value
`self.overlayingRoutes[localhostServerMethodPath]` returns a separate copy.

Reply:
You're totally right. Sorry, my bad.